### DISCUSSION TITLE
How to only let Biome format the file? vtsls reformats the file again

### DISCUSSION THREAD

**Discussion by Murderlon** [LazyVim/LazyVim#4397] [Q&A] [answered]
When using `extras.lang.typescript`, which installs the vtsls LSP, and using Biome both formatters run. Unfortunately, vtsls formats last and undos the required formatting from Biome. Sometimes Biome is the last formatter, it seems to be a race condition.

How can I make sure Biome formats last or alternatively disable formatting for vtsls when Biome is present?

**Comment by zaynOm** [comment]
can u check if you have prettier add to your extras

**Reply by Murderlon** [reply]
I have it because I have other projects with prettier but I have `vim.g.lazyvim_prettier_needs_config = true` and can confirm that prettier is not formatting here. `:ConformInfo` confirms it's vtsls:

```
Formatters for this buffer:
LSP: biome, vtsls
prettier unavailable: Condition failed
```

**Comment by dpetka2001** [comment, marked as answer]
You might want to take a look at the [config options here](https://github.com/stevearc/conform.nvim/blob/0ebe875d9c306f5fc829db38492ffff2a70d8e9d/doc/conform.txt#L20-L29). It seems that you're able to pass `format` options along with the desired formatter and also use a function to determine the formatters.

So, maybe something like
```lua
typescript = function(bufnr)
  if require("conform").get_formatter_info("biome", bufnr).available then
    return { "biome", lsp_format = "never" }
  else
    return { "whatever_other_formatter_you_want" }
  end
end
```

I've never done this myself, so can't be sure it'll work. Just test it out.

**Reply by Murderlon** [reply]
This seems to work! Thanks :)
